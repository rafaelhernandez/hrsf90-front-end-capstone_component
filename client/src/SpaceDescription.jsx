import React from 'react';
import { Glyphicon } from 'react-bootstrap';

const SpaceDescription = (props) => (
  <div className="div-description-space-description">
    <div className="div-description-open-fold">
      <a href="#">Read more about the space <Glyphicon glyph="chevron-down"/></a>
    </div>
    <div>
      {props.desc}
    </div>
    <div className="div-description-hide-fold">
      <a href="#">Hide <Glyphicon glyph="chevron-up"/></a>
    </div>
  </div>
);

export default SpaceDescription;
