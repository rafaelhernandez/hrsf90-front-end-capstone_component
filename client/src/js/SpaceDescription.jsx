import React from 'react';
import Summary from './Summary.jsx';
import { Glyphicon } from 'react-bootstrap';

const HiddenSpace = props => (
  <div>
    <Summary summary={props.desc.summary}></Summary>
    <div className="div-description-open-fold div-description-separator-margin">
      <a className="a-description-fold" onClick={props.onClick}>Read more about the space <Glyphicon glyph="chevron-down"/></a>
    </div>
  </div>
);


const Space = props => (
  <div>
    <div>
      <Summary summary={props.desc.summary}></Summary>
      <div className="div-description-space-header">
        The space
      </div>
      <div>
        {props.desc.space}
      </div>
      <div className="div-description-space-header">
        Guest access
      </div>
      <div>
        {props.desc.access}
      </div>
      <div className="div-description-space-header">
        Interaction with guests
      </div>
      <div>
        {props.desc.interaction}
      </div>
      <div className="div-description-space-header">
        Other things to note
      </div>
      <div>
        {props.desc.notes}
      </div>
      <div className="div-description-space-header">
        License or registration number
      </div>
      <div>
        <div>
          {props.desc.license}
        </div>
        <a className="a-description-fold" href="https://www.airbnb.com/help/article/1634/what-is-the-license-or-registration-number-field">Learn about this number</a>
      </div>
    </div>
    <div className="div-description-hide-fold">
      <p className="div-description-separator-margin">
        <a className="a-description-fold" onClick={props.onClick}>Hide <Glyphicon glyph="chevron-up"/></a>
      </p>
    </div>
  </div>
);

class SpaceDescription extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showSpace: false
    };
    this.showSpaceClick = this.showSpaceClick.bind(this);
    this.hideSpaceClick = this.hideSpaceClick.bind(this);
  }

  showSpaceClick() {
    this.setState({
      showSpace: true
    });
  }

  hideSpaceClick() {
    this.setState({
      showSpace: false
    });
  }

  render() {
    let space;
    if (this.state.showSpace) {
      space = <Space desc={this.props.desc} onClick={this.hideSpaceClick}></Space>;
    } else {
      space = <HiddenSpace desc={this.props.desc} onClick={this.showSpaceClick}></HiddenSpace>;
    }
    return (
      <div className="div-description-space-description">
        {space}
      </div>
    );
  }
}

export default SpaceDescription;
