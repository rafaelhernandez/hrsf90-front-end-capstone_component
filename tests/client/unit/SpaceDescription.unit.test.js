import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import SpaceDescription from '../../../client/src/js/SpaceDescription.jsx';

configure({ adapter: new Adapter() });

describe('SpaceDescription', () => {
  it('should render correctly', () => {
    const output = shallow(
      <SpaceDescription desc="Mock Long Space Description"></SpaceDescription>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
