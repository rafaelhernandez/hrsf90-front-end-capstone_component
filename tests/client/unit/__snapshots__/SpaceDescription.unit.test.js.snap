// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpaceDescription should render correctly 1`] = `
<div
  className="div-description-space-description"
>
  <HiddenSpace
    desc="Mock Long Space Description"
    onClick={[Function]}
  />
</div>
`;
